### PR TITLE
[KIWI-1259] Build an IPV Stub for the Banking CRI

### PR DESCRIPTION
**Pull request by hemavana**
### What changed

Adding the IPV stub code

### Why did it change

Banking CRI requirements

### Issue tracking

- [KIWI-1259](https://govukverify.atlassian.net/browse/KIWI-1259)

### PII logging

- [x] Verified that no PII data is being logged

[KIWI-1259]: https://govukverify.atlassian.net/browse/KIWI-1259?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: bav-ipv-stub/template.yaml [hunk 1]
```diff
@@ -0,0 +1,564 @@
...
+  PermissionsBoundary:
+    Description: "The ARN of the permissions boundary to apply when creating IAM roles. Mandatory."
+    Type: String
+    Default: "none"
+
+Mappings:
+  Configuration:
+    dev:
+      IPVStubID: "5C584572"
+      JWKSURI: "https://api-bav-cri-api.review-o.dev.account.gov.uk"
```
Comment (madankaruppiah) [inline bav-ipv-stub/template.yaml:45]: domain name for bav is "review-bar"

File: bav-ipv-stub/get-ssm-test-params.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#! /bin/bash
```
Comment (ViveakV) [inline bav-ipv-stub/get-ssm-test-params.sh:1]: Do we actually need this file I don't think we ever used it in F2F/CIC?

File: bav-ipv-stub/src/handlers/callback.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from "aws-lambda";
+
+export const handler = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  if (event.queryStringParameters?.error != null) {
+    return {
+      statusCode: 200,
+      body: "Authorisation Grant Failed. See logs for details",
```
Comment (ViveakV) [inline bav-ipv-stub/src/handlers/callback.ts:9]: [nit]: Can we just change to be something like "IPV Callback endpoint" so our journey doesn't end on a confusing page..

File: bav-ipv-stub/get-ssm-test-params.sh (hunk 2 above)
Comment (hemavana) [reply bav-ipv-stub/get-ssm-test-params.sh:1]: Deleted.

File: bav-ipv-stub/README.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+When you deploy this stack you will have access to 3 endpoints.
+
+To start a session use the start endpoint:
+```http 
+POST https://base.uri/stage/start
+Content-Type: application/json
+
+{
+    "target":"<OPTIONAL_TARGET_URI>",
```
Comment (Caitlin-cooling) [inline bav-ipv-stub/README.md:24]: Just using this for testing the /session endpoint and I've noticed that when I include a target I get a 500 - not sure if that's something to do with the DNS stuff?

image

Comment (hemavana) [reply bav-ipv-stub/README.md:24]: Works with the target, might be issue with the DNS urls;
Screenshot 2023-10-17 at 15 35 58

File: bav-ipv-stub/src/handlers/callback.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from "aws-lambda";
+
+export const handler = async (
```
Comment (ViveakV) [inline bav-ipv-stub/src/handlers/callback.ts:3]: Happy to leave this as is now but let's raise tech debt to make this response not so confusing 🙏

File: bav-ipv-stub/src/handlers/startBavCheck.ts [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+        ? overrides.shared_claims
+        : defaultClaims,
+    evidence_requested: overrides?.evidence_requested
+  };
+
+  if(!overrides?.evidence_requested){
+    delete payload.evidence_requested
+  }
+
+  console.log("Generate payload is"+JSON.stringify(payload));
```
Comment (ViveakV) [inline bav-ipv-stub/src/handlers/startBavCheck.ts:80]: Do we wanna leave this in here? Might clutter log

File: src/tests/utils/ApiTestSteps.ts [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+    }
+}
+
+export async function userInfoPost(accessToken?: any): Promise<any> {
+    const path = "/userInfo";
+    try{
+        const postRequest = await API_INSTANCE.post(path, null, { headers: { "Authorization": `${accessToken}` } });
+        return postRequest;
+    } catch(error: any){
+        console.log(`Error rrsponse from ${path} endpoint ${error}.`);
```
Comment (ViveakV) [inline src/tests/utils/ApiTestSteps.ts:65]: Does this not to return an error in the catch?

File: bav-ipv-stub/src/handlers/callback.ts (hunk 5 above)
Comment (hemavana) [reply bav-ipv-stub/src/handlers/callback.ts:3]: Yeah even i was not sure if i should update it as I wanted to understand where was this being used. Created a tech debt ticket to look into this later on: https://govukverify.atlassian.net/browse/KIWI-1345

**Review by ViveakV — APPROVED** [review]

**Pull request merged**